src/components/sidebar/index.tsx

```diff
@@ -102,12 +102,15 @@ const Sidebar = ({
   const [filterValue, setFilterValue] = useState('')
   const filteredMenuItems = getFilteredMenuItems(itemsWithMetadata, filterValue)
 
+  let backToLink
+  if (backToLinkProps) {
+    const { text, url } = backToLinkProps
+    backToLink = <SidebarBackToLink text={text} url={url} />
+  }
```

Comment:
Using this approach instead of `&&` or a ternary with `null` in the false-y branch.

Reply:
Neat! 🤩 

Non-blocking & very optional question: I'm curious to get your take on when we should reach for this pattern. My understanding of the current approaches:

- the `&&` approach is for when we're conditionally showing or not showing a component, and the condition being checked is known not to be a falsy value such as `0` or `""`
- the `{condition ? X : Y}` approach is for situations where we're either switching between X and Y components, or for show / hide situations where `condition` might be a falsy value such as `0` or `""` 

It seems like the `let` & `if` block approach might be useful for situations where there's some more complex logic or destructing we want to do when we do want to render the component. Does that seem accurate? 🤔

Reply:
Hopefully not misinterpreting @kendallstrautman's take on this from a PR I had a couple of days ago, but as I understand it we might just want to avoid `&&` altogether? She [linked a post](https://kentcdodds.com/blog/use-ternaries-rather-than-and-and-in-jsx#conclusion) that said even if you're sure the condition is a boolean, that using `&&` is abusing the operator.

As for ternaries, I don't think we should have multiline ternaries inside of JSX. If it's one line, that's not as difficult to read. But ternaries that span over several lines are hard to read and understand.

With both of these things in mind, that leaves using `let` and `if` before the `return` with markup. It's a little extra work, but it really does keep the markup clean and easy to read!